pkg/execution/executor/executor.go

```diff
@@ -182,7 +192,25 @@ func (e *executor) AddLifecycleListener(l execution.LifecycleListener) {
 
 // Execute loads a workflow and the current run state, then executes the
 // function's step via the necessary driver.
+//
+// If this function has a debounce config, this will return ErrFunctionDebounced instead
+// of an identifier as the function is not scheduled immediately.
 func (e *executor) Schedule(ctx context.Context, req execution.ScheduleRequest) (*state.Identifier, error) {
+	if req.Function.Debounce != nil && !req.PreventDebounce {
+		err := e.debouncer.Debounce(ctx, debounce.DebounceItem{
+			AccountID:       req.AccountID,
+			WorkspaceID:     req.WorkspaceID,
+			FunctionID:      req.Function.ID,
+			FunctionVersion: req.Function.FunctionVersion,
+			EventID:         req.Events[0].GetInternalID(),
+			Event:           req.Events[0].GetEvent(),
+		}, req.Function)
+		if err != nil {
+			return nil, err
+		}
+		return nil, ErrFunctionDebounced
```

Comment:
Non-blocker and follow up:
Probably better to make it _not_ return an error since this is intended behavior.

Reply:
Yeah.  I'm not the biggest fan of errors as status identifiers — ideally we'd have a way of responding with a status alongside an error.